From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: j-rockwell <john@llewkcor.com>
Date: Mon, 12 Oct 2026 14:54:38 -0700
Subject: [PATCH] Consistent knockback

A player hit by a legacy weapon is knocked from their 1.7.10 server
motion (MotionShadow), gets one velocity packet built from the knock,
and a plugin setVelocity resets that motion. The logic lives in
gg.hcfrevival.revival.combat; this patch only calls into it.

diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
index f584a9c763ec5375453f518390338b90729bb252..775cfe361c35e3a65f6429e185d416555ee8714e 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEntity.java
@@ -233,6 +233,7 @@ public abstract class CraftEntity implements org.bukkit.entity.Entity {
         // Paper end
         this.entity.setDeltaMovement(CraftVector.toVec3(velocity));
         this.entity.hurtMarked = true;
+        if (this.entity instanceof net.minecraft.server.level.ServerPlayer player) player.revivalMotionShadow.set(this.entity.getDeltaMovement()); // Revival - Consistent knockback
     }
 
     /**
